File: ShowAndHideUIElemnts/app/src/main/java/com/showandhideuielemnts/MainActivity.java
package com.showandhideuielemnts;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {

    TextView helloWorld;

    public void show(View view) {
        helloWorld.setVisibility(View.VISIBLE);
    }

    public void hide(View view) {
        helloWorld.setVisibility(View.INVISIBLE);
    }


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        helloWorld = findViewById(R.id.helloWorldTextView);


    }
}

File: Udacity Course/Dice Roller/app/src/main/java/com/diceroller/MainActivity.kt
package com.diceroller

import android.os.Bundle
import android.widget.Button
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import kotlinx.android.synthetic.main.activity_main.*
import java.util.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val rollButton = roll_button as Button
        rollButton.setOnClickListener{
            // Toast.makeText(this, "button clicked", Toast.LENGTH_LONG).show()
            rollDice()
        }
    }

    private fun rollDice() {
        val resultText = result_text as TextView

        val randomInt = Random().nextInt(6) + 1
        resultText.text = randomInt.toString()
    }
}
